qt/src/SourceManager.cc

```diff
@@ -44,6 +54,32 @@ QDataStream& operator>>(QDataStream& ds, Source*&) {
 	return ds;
 }
 
+Source::Source(const QString& _path, const QString& _displayname, bool _isTemp)
+				: path(_path), displayname(_displayname), isTemp(_isTemp) {
+	// Check whether input PDF file is encrypted
+	if(path.endsWith(".pdf", Qt::CaseInsensitive)) {
+		std::unique_ptr<Poppler::Document> document(Poppler::Document::load(path));
+		if(document && document->isLocked()) {
+			bool ok = false;
+			QString text;
+			while(true) {
+				text = QInputDialog::getText(MAIN, _("Enter password for the PDF file"),
+											 _("Password:"), QLineEdit::Password,
+											 "", &ok);
+				if(!ok) {
+					throw std::invalid_argument("Locked PDF: skip file.");
+				}
+				if(!document->unlock(text.toLocal8Bit(), text.toLocal8Bit())) {
```

Comment:
I checked the poppler source code, and looks like the return value of `unlock` is whether the document is still locked. So `false` means that the document is not locked after the unlock attempt, hence success... So yes, ! is correct.

Reply:
:+1: